test/dataframe.jl

```diff
@@ -325,25 +325,48 @@ module TestDataFrame
     df = DataFrame(a=Union{Int, Missing}[1, 2, 3], b=Union{Float64, Missing}[3.0, 4.0, 5.0])
     @test deleterows!(df, [2, 3]) === df
     @test df == DataFrame(a=[1], b=[3.0])
-
-    # describe
-    #suppress output and test that describe() does not throw
-    devmissing = Compat.Sys.isunix() ? "/dev/null" : "nul"
-    open(devmissing, "w") do f
-        @test nothing == describe(f, DataFrame(a=[1, 2], b=Any["3", missing]))
-        @test nothing ==
-              describe(f, DataFrame(a=Union{Int, Missing}[1, 2],
-                                    b=["3", missing]))
-        @test nothing ==
-              describe(f, DataFrame(a=CategoricalArray([1, 2]),
-                                    b=CategoricalArray(["3", missing])))
-        @test nothing == describe(f, [1, 2, 3])
-        @test nothing == describe(f, [1, 2, 3])
-        @test nothing == describe(f, CategoricalArray([1, 2, 3]))
-        @test nothing == describe(f, Any["1", "2", missing])
-        @test nothing == describe(f, ["1", "2", missing])
-        @test nothing == describe(f, CategoricalArray(["1", "2", missing]))
-    end
+    
+    # Test that the describe output handles all values and missings properly
+    # construct the output dataframe
+    struct MySuperLongNameForAStructIsThisToolong end
+    @testset "describe(df)" begin
+        variable = Vector{Any}([:number, :number_missing, :non_number, :non_number_missing, :long_struct])
```

Comment:
Just create the `DataFrame` directly, no need to store columns in temporary objects. Also I don't think we care about the type of the vector, it will be chosen automatically. Same below.